### DISCUSSION TITLE
Lightweight data loader

### DISCUSSION THREAD

**Discussion by chrisflesher** [jax-ml/jax#16785] [General] [answered]
I was reading the notes about data loading and was surprised that both options (tensorflow-datasets and pytorch) seem to require 1-3GB installs! Is there a minimal data loader that doesn't require so much disk space?

**Comment by hawkinsp** [comment]
I don't think `tensorflow_datasets` depends on `tensorflow` any more, so the wheel download size is more like 10MB.

There are other libraries (e.g., https://github.com/google/grain) that you may want to look into, although the public docs on `grain` seem minimal.

**Reply by chrisflesher** [reply]
I tried running the tensorflow-datasets example but without tensorflow it crashed when loading the MINST dataset. Maybe for custom datasets this might work? I don't have that much experience using it.

May be interested in grain but couldn't find any documentation...

**Comment by adam-hartshorne** [comment]
Have you tried jax_dataloader ?
https://github.com/BirkhoffG/jax-dataloader

**Reply by chrisflesher** [reply]
Thanks, will look at this.

**Comment by moskomule** [comment, marked as answer]
huggingface's datasets now supports jax!
https://huggingface.co/docs/datasets/use_with_jax

**Reply by chrisflesher** [reply]
Wow, this seems like a really great option for what I'm doing. Lightweight, fast, and widely used.
